rust/src/protocol_types/fixed_tx.rs

```diff
@@ -111,4 +137,22 @@ impl FixedTransaction {
     pub fn raw_auxiliary_data(&self) -> Option<Vec<u8>> {
         self.auxiliary_bytes.clone()
     }
+
+    pub fn sign_and_add_vkey_signature(&mut self, private_key: &PrivateKey) -> Result<(), JsError> {
+        let vkey_witness = make_vkey_witness(&self.tx_hash, private_key);
+        self.witness_set.add_vkey_witness(vkey_witness);
+        Ok(())
+    }
```

Comment:
@lisicky , it would be great to have a variant of this API that accepts and adds an existing `vkey_witness` , remember we have hardware keys and some other cases in which the signing is done separately and we never get access to the private key itself.

Reply:
Done